assets/js/index.js

```diff
@@ -146,15 +162,47 @@ function bindDeleteEvent() {
   });
 }
 
-function bindUpdateEvent() {
-  const checkboxs = Array.from(document.getElementsByClassName("checkbox"));
-  checkboxs.forEach(checkbox => {
-    checkbox.addEventListener("change", function (event) {
+function bindUpdateIsDoneEvent() {
+  const taskItemCheckboxes = document.querySelectorAll(".task-item-checkbox");
+  taskItemCheckboxes.forEach(taskItemCheckbox => {
+    taskItemCheckbox.addEventListener("change", function (event) {
       updateStatus(event.target.parentNode.id, this.checked);
     })
   });
 }
 
+async function bindUpdateTextEvent() {
+  const editIcons = document.querySelectorAll(".edit-icon");
+  editIcons.forEach(editIcon => {
+    editIcon.addEventListener("click", function (event) {
+      const { id: taskId } = event.target.parentNode;
+      const todoContainer = document.getElementById(`todo-container-${taskId}`);
+      const todoText = todoContainer.innerText;
+      todoContainer.innerHTML = `<input id="input-${taskId}" type="text" value="${todoText}"/>
+      <span id="check-icon-${taskId}" class="fa-solid fa-check check-icon">23</span>`
+      const checkIcon = document.getElementById(`check-icon-${taskId}`);
+      const input = document.getElementById(`input-${taskId}`);
```

Comment:
Apply this and change the code accordingly.
```suggestion
      const newValue = document.getElementById(`input-${taskId}`).value.trim();
```